core/dbt/task/init.py

```diff
@@ -257,7 +261,15 @@ def ask_for_adapter_choice(self) -> str:
     def get_valid_project_name(self) -> str:
         """Returns a valid project name, either from CLI arg or user prompt."""
         name = self.args.project_name
-        while not ProjectName.is_valid(name):
+        internal_package_names = {GLOBAL_PROJECT_NAME}
+        available_adapters = list(_get_adapter_plugin_names())
+        for adapter_name in available_adapters:
+            try:
+                internal_package_names.update(get_adapter_package_names(adapter_name))
+            # When no adapter plugins found
+            except dbt.exceptions.InternalException:
+                pass
```

Comment:
Since `get_valid_project_name` is only run during the `init` task, the adapters have not yet been loaded into the adapter factory, meaning that `get_adpater_package_names` will return an empty array, even if the adapter does contain packages!

I will create a tech debt ticket to look into whether it is worth loading adapters earlier, but for now, modifying this code to assume that packages maintained by dbt Labs will be prefixed with `dbt_` is a safe assumption:
```suggestion
        for adapter_name in available_adapters:
            internal_package_names.update(f"dbt_{adapter_name}")
```